mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -34,8 +36,25 @@ import org.apache.spark.util.collection.OpenHashMap
 /**
  * Base trait for [[StringIndexer]] and [[StringIndexerModel]].
  */
-private[feature] trait StringIndexerBase extends Params with HasInputCol with HasOutputCol
-    with HasHandleInvalid {
+private[feature] trait StringIndexerBase extends Params with HasInputCol with HasOutputCol {
+  val SKIP_UNSEEN_LABEL: String = "skip"
```

Comment:
Constants like this should go in a static object class.  I'd also like to keep them private for now.